keanu-project/src/main/java/io/improbable/keanu/util/io/MIRLoader.java

```diff
@@ -0,0 +1,69 @@
+package io.improbable.keanu.util.io;
+
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.mir.MIR;
+import io.improbable.mir.SavedBayesNet;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class MIRLoader extends ProtobufLoader {
+
+    @Override
+    public BayesianNetwork loadNetwork(InputStream input) throws IOException {
+        MIR.Model model = MIR.Model.parseFrom(input);
+        checkModelIsCompatible(model);
+
+        return extractGraphAndLoadNetwork(model);
+    }
+
+    private static void checkModelIsCompatible(MIR.Model model) {
+        checkProperties(model);
+        checkEntryPoint(model);
+        checkFunction(model);
+    }
+
+    private static void checkFunction(MIR.Model model) {
+        if (model.getFunctionsByNameMap().get(MIRSaver.ENTRY_POINT_NAME) == null) {
+            throw new IllegalArgumentException("Expected Entry Point not found");
+        }
+    }
+
+    private static void checkEntryPoint(MIR.Model model) {
+        if (!model.getEntryPointName().equals(MIRSaver.ENTRY_POINT_NAME)) {
+            throw new IllegalArgumentException("Keanu only supports loading Keanu generated Graphs");
```

Comment:
Do we need this rule? Provided that the graph being loaded is a BN, it shouldn't matter where the graph came from.

Reply:
We certainly could leave it configurable, but until we start adding features I like that we're being explicit that you should only really use it to save and load stuff keanu itself generated.